### ISSUE TITLE
Payment Reference

### ISSUE THREAD

**Issue by ligi** [D-D-Agency/woop-pay#4]
Interesting project! Could be a nice alternative for [web3modal checkout](https://github.com/WalletConnect/web3modal-checkout) that we use in in the [pretix payment plugin](https://github.com/esPass/pretix-eth-payment-plugin)
Wondering of you plan to add a (signed) payment reference or any other solution to the use-case that you expect a lot of payments on one address for different invoices.

**Comment by alessandromaci** [comment]
hi @ligi, apologies for the late reply.

Can you specify what you mean by "add a payment reference"?

**Comment by ligi** [comment]
some signed message from the sending account on the purpose of the tx. E.g. a invoice number.

**Comment by alessandromaci** [comment]
Yes, It is something we want to explore. Messaging between requester/payer and automated payment invoice generator are in our bucket list.

What specific use case do you have that requires an invoice number?

**Comment by ligi** [comment]
Imagine you want to do an online-shop that creates multiple payment requests - then you need to know which payment requests where confirmed and correlate the order to the payment request.
